apps/dashboard/src/@/api/support.ts

```diff
@@ -0,0 +1,354 @@
+"use server";
+import "server-only";
+import { NEXT_PUBLIC_THIRDWEB_API_HOST } from "@/constants/public-envs";
+import { getAuthToken, getAuthTokenWalletAddress } from "./auth-token";
+
+export interface SupportTicket {
+  id: string;
+  status: "needs_response" | "in_progress" | "on_hold" | "closed" | "resolved";
+  createdAt: string;
+  updatedAt: string;
+  messages?: SupportMessage[];
+}
+
+interface SupportMessage {
+  id: string;
+  content: string;
+  createdAt: string;
+  timestamp: string;
+  author?: {
+    name: string;
+    email: string;
+    type: "user" | "customer";
+  };
+}
+
+interface CreateSupportTicketRequest {
+  message: string;
+  teamSlug: string;
+  title: string;
+  conversationId?: string;
+}
+
+interface SendMessageRequest {
+  ticketId: string;
+  teamSlug: string;
+  message: string;
+}
+
+export async function getSupportTicketsByTeam(
+  teamSlug: string,
+  authToken?: string,
+): Promise<SupportTicket[]> {
+  if (!teamSlug) {
+    throw new Error("Team slug is required to fetch support tickets");
+  }
+
+  const token = authToken || (await getAuthToken());
+  if (!token) {
+    throw new Error("No auth token available");
+  }
+
+  // URL encode the team slug to handle special characters like #
+  const encodedTeamSlug = encodeURIComponent(teamSlug);
+  const apiUrl = `${NEXT_PUBLIC_THIRDWEB_API_HOST}/v1/teams/${encodedTeamSlug}/support-conversations/list`;
+
+  // Build the POST payload according to API spec
+  const payload = {
+    limit: 50,
+    descending: true,
+  };
+
+  const response = await fetch(apiUrl, {
+    body: JSON.stringify(payload),
+    cache: "no-store",
+    headers: {
+      Accept: "application/json",
+      "Accept-Encoding": "identity",
+      Authorization: `Bearer ${token}`,
+      "Content-Type": "application/json",
+    },
+    method: "POST",
+  });
+
+  if (!response.ok) {
+    const errorText = await response.text();
+    throw new Error(`API Server error: ${response.status} - ${errorText}`);
+  }
+  const data: { data?: SupportTicket[] } = await response.json();
+  const conversations = data.data || [];
+  return conversations;
+}
+
+interface RawSupportMessage {
+  id: string;
+  text?: string;
+  timestamp?: string;
+  createdAt?: string;
+  isPrivateNote?: boolean;
+  sentByUser?: {
+    name: string;
+    email: string;
+    isExternal: boolean;
+  };
+  // Add any other fields you use from the API
+}
+
+export async function getSupportTicket(
+  ticketId: string,
+  teamSlug: string,
+  authToken?: string,
+): Promise<SupportTicket | null> {
+  if (!ticketId || !teamSlug) {
+    throw new Error("Ticket ID and team slug are required");
+  }
+
+  const token = authToken || (await getAuthToken());
+  if (!token) {
+    throw new Error("No auth token available");
```

Comment:
Don't throw error in server action, This error won't shown up in frontend in production build. You'll only be able to see this error in dev. Return `{ data: ... } | { error: ''}` instead